spec/services/products_renderer_spec.rb

```diff
@@ -0,0 +1,117 @@
+require 'spec_helper'
+
+describe ProductsRenderer do
+  let(:distributor) { create(:distributor_enterprise) }
+  let(:order_cycle) { create(:simple_order_cycle, distributors: [distributor]) }
+  let(:exchange) { order_cycle.exchanges.to_enterprises(distributor).outgoing.first }
+  let(:customer) { create(:customer) }
+  let(:pr) { ProductsRenderer.new(distributor, order_cycle, customer) }
+
+  describe "sorting" do
+    let(:t1) { create(:taxon) }
+    let(:t2) { create(:taxon) }
+    let!(:p1) { create(:product, name: "abc", primary_taxon_id: t2.id) }
+    let!(:p2) { create(:product, name: "def", primary_taxon_id: t1.id) }
+    let!(:p3) { create(:product, name: "ghi", primary_taxon_id: t2.id) }
+    let!(:p4) { create(:product, name: "jkl", primary_taxon_id: t1.id) }
+
+    before do
+      exchange.variants << p1.variants.first
+      exchange.variants << p2.variants.first
+      exchange.variants << p3.variants.first
+      exchange.variants << p4.variants.first
+    end
+
+    it "sorts products by the distributor's preferred taxon list" do
+      allow(distributor).to receive(:preferred_shopfront_taxon_order) { "#{t1.id},#{t2.id}" }
+      products = pr.send(:products)
+      expect(products).to eq([p2, p4, p1, p3])
+    end
+
+    it "alphabetizes products by name when taxon list is not set" do
+      allow(distributor).to receive(:preferred_shopfront_taxon_order) { "" }
+      products = pr.send(:products)
+      expect(products).to eq([p1, p2, p3, p4])
+    end
+  end
+
+  context "JSON tests" do
+    let(:product) { create(:product) }
+    let(:variant) { product.variants.first }
+
+    before do
+      exchange.variants << variant
+    end
+
+    it "only returns products for the current order cycle" do
+      expect(pr.products_json).to include product.name
+    end
+
+    it "doesn't return products not in stock" do
+      variant.update_attribute(:on_demand, false)
+      variant.update_attribute(:on_hand, 0)
+      expect(pr.products_json).not_to include product.name
+    end
+
+    it "strips html from description" do
+      product.update_attribute(:description, "<a href='44'>turtles</a> frogs")
+      json = pr.products_json
+      expect(json).to include "frogs"
+      expect(json).not_to include "<a href"
+    end
+
+    it "returns price including fees" do
+      # Price is 19.99
+      allow_any_instance_of(OpenFoodNetwork::EnterpriseFeeCalculator).
```

Comment:
:warning: the `any_instance` are not recommended as lead to brittle tests. It might make things easy in here but pretty please, let's not lower the guard :pray: 

Even RSpec docs do not recommend them https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance

> This feature is sometimes useful when working with legacy code, though in general we discourage its use for a number of reasons:

Reply:
This is also an old test that was just copied across when the `ProductsRenderer` namespace was changed. 

What would you like it to be?

Reply:
These can be solved by injection test doubles in the right place so that in this case it's not `any` `EnterpriseFeeCalculator` but the one you expect. I'm not asking to change it but just to be aware of the bad practice.